x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts

```diff
@@ -23,75 +24,100 @@ import {
   IndexFieldsStrategyRequest,
   BeatFields,
 } from '../../../common/search_strategy/index_fields';
+import { StartPlugins } from '../../types';
 
-const apmIndexPattern = 'apm-*-transaction*';
-const apmDataStreamsPattern = 'traces-apm*';
+// SOMEONE EXPLAIN WHY I SHOULDN'T DELETE?
+// const apmIndexPattern = 'apm-*-transaction*';
+// const apmDataStreamsPattern = 'traces-apm*';
 
-export const indexFieldsProvider = (): ISearchStrategy<
-  IndexFieldsStrategyRequest,
-  IndexFieldsStrategyResponse
-> => {
+export const indexFieldsProvider = (
+  getStartServices: StartServicesAccessor<StartPlugins>
+): ISearchStrategy<IndexFieldsStrategyRequest, IndexFieldsStrategyResponse> => {
   // require the fields once we actually need them, rather than ahead of time, and pass
   // them to createFieldItem to reduce the amount of work done as much as possible
   // eslint-disable-next-line @typescript-eslint/no-var-requires
   const beatFields: BeatFields = require('../../utils/beat_schema/fields').fieldsBeat;
 
   return {
-    search: (request, options, deps) => from(requestIndexFieldSearch(request, deps, beatFields)),
+    search: (request, options, deps) =>
+      from(requestIndexFieldSearch(request, deps, beatFields, getStartServices)),
   };
 };
 
+export const findExistingIndices = async (
+  indices: string[],
+  esClient: ElasticsearchClient
+): Promise<boolean[]> =>
+  Promise.all(
+    indices
+      .map(async (index) => {
+        const searchResponse = await esClient.search({
+          index,
+          body: { query: { match_all: {} }, size: 0 },
+        });
+        return get(searchResponse, 'body.hits.total.value', 0) > 0;
+      })
+      .map((p) => p.catch((e) => false))
+  );
+
 export const requestIndexFieldSearch = async (
   request: IndexFieldsStrategyRequest,
-  { esClient }: SearchStrategyDependencies,
-  beatFields: BeatFields
+  { savedObjectsClient, esClient }: SearchStrategyDependencies,
+  beatFields: BeatFields,
+  getStartServices: StartServicesAccessor<StartPlugins>
 ): Promise<IndexFieldsStrategyResponse> => {
   const indexPatternsFetcherAsCurrentUser = new IndexPatternsFetcher(esClient.asCurrentUser);
   const indexPatternsFetcherAsInternalUser = new IndexPatternsFetcher(esClient.asInternalUser);
 
   const dedupeIndices = dedupeIndexName(request.indices);
-
-  const responsesIndexFields = await Promise.all(
-    dedupeIndices
-      .map(async (index) => {
-        if (
-          request.onlyCheckIfIndicesExist &&
-          (index.includes(apmIndexPattern) || index.includes(apmDataStreamsPattern))
```

Comment:
this does not make sense to me. if the `request.onlyCheckIfIndicesExist` why don't we only check if indices exist no matter what patterns the data view contains? i clicked through those issues but still dont understand this logic. maybe if it was an `||` condition

Reply:
@ogupte @semd I see you in the git blame, can you speak to this at all?

Reply:
paired with @XavierM to cover all these cases